File: HNOJ/BANH_TRUNG_THU.py
n, m = map(int, input().split())
a = input().split()
ans = 0
res = [int(i) for i in a]
res.sort()
for i in range(0, n):
    if (m > 0):
        m -= int(res[i])
        ans++;
        
while (m >= int(res[0])):
    m -= int(res[0])
    ans++;
    
print(ans)

File: HNOJ/BANH_TRUNG_THU.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
int main(){
    ll n,m,ans=0; cin>>n>>m;
    ll a[n];
    for(int i=0; i<n; i++) cin>>a[i]; 
    sort(a,a+n);
    for(int i=0; i<n; i++){
        if(m>0){ m-=a[i]; ans++; }
    }
    while(m>=a[0]){ m-=a[0]; ans++; }
    cout<<ans;
}